golang/cosmos/x/swingset/genesis.go

```diff
@@ -169,32 +190,72 @@ type swingStoreGenesisEventHandler struct {
 	snapshotHeight uint64
 	swingStore     sdk.KVStore
 	hasher         hash.Hash
+	exportMode     string
 }
 
 func (eventHandler swingStoreGenesisEventHandler) OnExportStarted(height uint64, retrieveSwingStoreExport func() error) error {
 	return retrieveSwingStoreExport()
 }
 
 func (eventHandler swingStoreGenesisEventHandler) OnExportRetrieved(provider keeper.SwingStoreExportProvider) error {
-	if eventHandler.snapshotHeight != provider.BlockHeight {
+	if !((eventHandler.exportMode == keeper.SwingStoreArtifactModeDebug && eventHandler.snapshotHeight == 0) ||
+		eventHandler.snapshotHeight == provider.BlockHeight) {
 		return fmt.Errorf("snapshot block height (%d) doesn't match requested height (%d)", provider.BlockHeight, eventHandler.snapshotHeight)
 	}
 
+	artifactsEnded := false
+
+	var getExportDataReader = func() (agoric.KVEntryReader, error) {
+		exportDataIterator := eventHandler.swingStore.Iterator(nil, nil)
+		kvReader := agoric.NewKVIteratorReader(exportDataIterator)
+		eventHandler.hasher.Reset()
+		encoder := json.NewEncoder(eventHandler.hasher)
+		encoder.SetEscapeHTML(false)
+
+		return agoric.NewKVHookingReader(kvReader, func(entry agoric.KVEntry) error {
+			return encoder.Encode(entry)
+		}, func() error {
+			return nil
+		}), nil
+	}
+
 	artifactsProvider := keeper.SwingStoreExportProvider{
```

Comment:
In the case of debug mode (and only in this case), the export artifact should capture the blockHeight of the swing-store, so we don't completely lose track of it (we unfortunately don't have a better place to stash it)

```go
if eventHandler.exportMode == keeper.SwingStoreArtifactModeDebug {
  artifactProvider.BlockHeight = provider.BlockHeight
}
```